test/contracts/core/scenario/twapRebalancingSetScenarios.spec.ts

```diff
@@ -0,0 +1,417 @@
+    },
+    components: {
+      component1Price: ether(200),
+      component2Price: ether(0.021),
+      component1Decimals: 18,
+      component2Decimals: 8,
+    },
+    auction: {
+      chunkSize: ether(1000000),
+      chunkAuctionPeriod: new BigNumber(0),
+      expectedAuctions: 6,
+    }
+  },
+  {
+    name: 'WBTC 100% WETH to 50/50% wBTC/cUSDC #706',
+    rebalancingSet: {
+      unitShares: new BigNumber(296562),
+      naturalUnit: new BigNumber(100000000),
+      supply: new BigNumber('37566900000000000000')
+    },
+    currentSet: {
+      components: ['component1'], // cUSDC
+      units: [new BigNumber(166187652)],
+      naturalUnit: new BigNumber(1000000000000),
+    },
+    nextSet: {
+      components: ['component1', 'component2'], // cUSDC, WBTC
+      units: [new BigNumber(41946670), new BigNumber(100)],
+      naturalUnit: new BigNumber(1000000000000),
+    },
+    components: {
+      component1Price: ether(0.021),
+      component2Price: ether(9000),
+      component1Decimals: 8,
+      component2Decimals: 8,
+    },
+    auction: {
+      chunkSize: ether(1000000),
+      chunkAuctionPeriod: new BigNumber(0),
+      expectedAuctions: 1,
+    }
+  }
+];
+
+
+contract('RebalancingSetV3 - TWAPLiquidator Scenarios', accounts => {
+  const [
+    deployerAccount,
+    managerAccount,
+    otherAccount,
+    feeRecipient,
+  ] = accounts;
+
+  let rebalancingSetToken: RebalancingSetTokenV3Contract;
+
+  let liquidator: TWAPLiquidatorContract;
+
+  let name: string = 'liquidator';
+  let auctionPeriod: BigNumber = ONE_HOUR_IN_SECONDS.mul(4);
+  let rangeStart: BigNumber = new BigNumber(1);
+  let rangeEnd: BigNumber = new BigNumber(21);
+
+  let setup: RebalanceTestSetup;
+
+  const coreHelper = new CoreHelper(deployerAccount, deployerAccount);
+  const erc20Helper = new ERC20Helper(deployerAccount);
+  const rebalancingHelper = new RebalancingSetV3Helper(
+    deployerAccount,
+    coreHelper,
+    erc20Helper,
+    blockchain
+  );
+  const oracleHelper = new OracleHelper(deployerAccount);
+  const valuationHelper = new ValuationHelper(deployerAccount, coreHelper, erc20Helper, oracleHelper);
+  const liquidatorHelper = new LiquidatorHelper(deployerAccount, erc20Helper, valuationHelper);
+
+  before(async () => {
+    ABIDecoder.addABI(CoreMock.abi);
+  });
+
+  after(async () => {
+    ABIDecoder.removeABI(CoreMock.abi);
+  });
+
+  beforeEach(async () => {
+    blockchain.saveSnapshotAsync();
+
+    setup = new RebalanceTestSetup(deployerAccount);
+  });
+
+  afterEach(async () => {
+    blockchain.revertAsync();
+  });
+
+  describe(`${scenarios[0].name}`, async () => {
+    it('should successfully complete', async () => {
+      await runScenario(scenarios[0]);
+
+      const newRebalanceState = await setup.rebalancingSetToken.rebalanceState.callAsync();
+      expect(newRebalanceState).to.be.bignumber.equal(SetUtils.REBALANCING_STATE.DEFAULT);
+    });
+  });
+
+  describe(`${scenarios[1].name}`, async () => {
```

Comment:
What are the most high leverage things we can test here? To me it seems like `rebalanceState`, currentSet`, and most importantly `unitShares`